app/src/main/java/com/github/se/eventradar/viewmodel/ChatViewModel.kt

```diff
@@ -0,0 +1,150 @@
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.update
+
+@HiltViewModel
+class ChatViewModel
+@Inject
+constructor(
+    private val messagesRepository: IMessageRepository,
+    private val userRepository: IUserRepository,
+) : ViewModel() {
+  private val _uiState = MutableStateFlow(ChatUiState())
+  val uiState: StateFlow<ChatUiState> = _uiState
+
+  fun onMessageBarInputChange(newInput: String) {
+    _uiState.update { currentState -> currentState.copy(messageBarInput = newInput) }
+  }
+
+  // TO DO: Complete VM functions for chat screen
+  /**
+   * // TO DO: Copied from MessagesViewModel.kt init { viewModelScope.launch { _uiState.update { val
+   * userId = userRepository.getCurrentUserId()
+   *
+   * if (userId is Resource.Success) { it.copy(userId = userId.data) } else { Log.d(
+   * "MessagesViewModel", "Error getting user ID: ${(userId as
+   * Resource.Failure).throwable.message}") it.copy(userId = null) } }
+   *
+   * if (_uiState.value.userId != null) { getMessages() getFriends() } } }
+   *
+   * fun onChatInputFocusChanged(state: MutableStateFlow<ChatUiState> = _uiState) { state.value =
+   * state.value.copy(isChatInputFocus = !state.value.isChatInputFocus) }
+   *
+   * fun onMessageSend() { val message = _uiState.value.messageBarInput if (message.isNotBlank()) {
+   * sendMessage(message) } insertMessageToFirebase( senderId, messageContent ) }
+   *
+   * fun insertMessageToFirebase( chatRoomUUID: String, messageContent: String, registerUUID:
+   * String, oneSignalUserId: String ) { viewModelScope.launch {
+   * chatScreenUseCases.insertMessageToFirebase( chatRoomUUID, messageContent, registerUUID,
+   * oneSignalUserId ).collect { response -> when (response) { is Response.Loading -> {
+   * messageInserted.value = false } is Response.Success -> { messageInserted.value = true } is
+   * Response.Error -> {} } } } }
+   *
+   * fun loadMessagesFromFirebase(chatRoomUUID: String, opponentUUID: String, registerUUID: String)
+   * { viewModelScope.launch { chatScreenUseCases.loadMessageFromFirebase(chatRoomUUID,
+   * opponentUUID, registerUUID) .collect { response -> when (response) { is Response.Loading -> {}
+   * is Response.Success -> { messages = listOf() for (i in response.data) { if (i.profileUUID ==
+   * opponentUUID) { messages = messages + MessageRegister(i, true) //Opponent Message } else {
+   * messages = messages + MessageRegister(i, false) //User Message }
+   *
+   * } messagesLoadedFirstTime.value = true } is Response.Error -> {} } } } }
+   *
+   * fun loadOpponentProfileFromFirebase(opponentUUID: String) { viewModelScope.launch {
+   * chatScreenUseCases.opponentProfileFromFirebase(opponentUUID).collect { response -> when
+   * (response) { is Response.Loading -> {} is Response.Success -> {
+   * opponentProfileFromFirebase.value = response.data } is Response.Error -> {} } } } }
+   */
+}
+
+data class ChatUiState(
+    //    val userId: String? = "1",
```

Comment:
Is this commented section still useful? if not, consider deleting it.

Reply:
Correct, I forgot to delete